go-controller/pkg/util/pod_annotation.go

```diff
@@ -491,10 +507,62 @@ func AllocatePodAnnotation(
 	podAnnotation *PodAnnotation,
 	err error) {
 
+	// no id allocation
+	var idAllocator idAllocator
+
 	allocateToPodWithRollback := func(pod *v1.Pod) (*v1.Pod, func(), error) {
 		var rollback func()
 		pod, podAnnotation, rollback, err = allocatePodAnnotationWithRollback(
 			ipAllocator,
+			idAllocator,
+			netInfo,
+			pod,
+			network,
+			reallocateIP)
+		return pod, rollback, err
+	}
+
+	err = UpdatePodWithRetryOrRollback(
+		podLister,
+		kube,
+		pod,
+		allocateToPodWithRollback,
+	)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return pod, podAnnotation, nil
+}
+
+// AllocatePodAnnotationWithID allocates the PodAnnotation which includes IPs, a
+// mac address, routes, gateways and an ID. Returns the allocated pod annotation
+// and the updated pod. Returns a nil pod and the existing PodAnnotation if no
+// updates are warranted to the pod.
+//
+// The allocation can be requested through the network selection element or
+// derived from the allocator provided IPs. If the requested IPs cannot be
+// honored, a new set of IPs will be allocated unless reallocateIP is set to
+// false.
+func AllocatePodAnnotationWithID(
```

Comment:
Renamed to `AllocatePodAnnotationWithTunnelID`. I don't really want to be more specific than that just in case is used for other things in the future.